### PR TITLE
Add --rune flag to resume command to enable resuming a specific etching

### PR DESCRIPTION
**Pull request by ldiego08**
Adds a new `--rune <RUNE>` flag to allow resuming a specific pending etching. This is useful in automated scenarios (if anyone else is into that kind of stuff) where you'd like to etch something, suspend (perhaps via `--no-wait` from #3526), do something else, and then resume after the 6 confirmations.

### CODE REVIEW

File: src/subcommand/wallet/resume.rs [hunk 1]
```diff
@@ -8,6 +8,9 @@ pub struct ResumeOutput {
 pub(crate) struct Resume {
   #[arg(long, help = "Don't broadcast transactions.")]
   pub(crate) dry_run: bool,
+
```
Comment (raphjaph) [inline src/subcommand/wallet/resume.rs:11]:
```suggestion
```

File: src/subcommand/wallet/resume.rs [hunk 2]
```diff
@@ -8,6 +8,9 @@ pub struct ResumeOutput {
...
+  #[arg(long, help = "Rune commitment to resume.")]
+  pub(crate) commitment: Option<String>,
```
Comment (raphjaph) [inline src/subcommand/wallet/resume.rs:13]:
```suggestion
  pub(crate) rune: Option<SpacedRune>,
```

File: src/subcommand/wallet/resume.rs [hunk 3]
```diff
@@ -18,7 +21,26 @@ impl Resume {
         break;
       }
 
-      for (rune, entry) in wallet.pending_etchings()? {
+      let mut pending_etchings = Vec::new();
+      let commitment = self.commitment.clone();
+
+      if let Some(commitment) = commitment {
```
Comment (raphjaph) [inline src/subcommand/wallet/resume.rs:27]:
```suggestion
      if let Some(rune) = self.rune {
```

**Review by raphjaph — CHANGES_REQUESTED** [review]
This is good feature.

We should probably resume it by rune name. Another feature that would improve this would be a command to list all pending etchings, so it is easier to know what can be resumed. Probably best to do another PR for that.

**Comment by ldiego08** [comment]
> This is good feature.
> 
> We should probably resume it by rune name. Another feature that would improve this would be a command to list all pending etchings, so it is easier to know what can be resumed. Probably best to do another PR for that.

Thanks for the feedback, @raphjaph - I'll get these changes in and create a separate PR for listing pending etchings.

**Comment by ldiego08** [comment]
Thanks for the feedback, @raphjaph! Requested changes have been applied. Will add the command to show pending etchings on a separate PR.

File: src/subcommand/wallet/resume.rs [hunk 4]
```diff
@@ -18,7 +20,26 @@ impl Resume {
...
 
-      for (rune, entry) in wallet.pending_etchings()? {
+      let mut pending_etchings = Vec::new();
+      let spaced_rune = self.rune.clone();
+
+      if let Some(spaced_rune) = spaced_rune {
+        let pending_etching = wallet
+          .pending_etchings()?
+          .into_iter()
+          .find(|(rune, _)| *rune == spaced_rune.rune);
```
Comment (raphjaph) [inline src/subcommand/wallet/resume.rs:30]:
```suggestion
          .load_etching(spaced_rune.rune)?;
         
```

File: src/subcommand/wallet/resume.rs [hunk 5]
```diff
@@ -18,7 +20,26 @@ impl Resume {
         break;
       }
 
-      for (rune, entry) in wallet.pending_etchings()? {
+      let mut pending_etchings = Vec::new();
+      let spaced_rune = self.rune.clone();
+
+      if let Some(spaced_rune) = spaced_rune {
```
Comment (raphjaph) [inline src/subcommand/wallet/resume.rs:26]:
```suggestion
     let pending_etchings = if let Some(spaced_rune) = spaced_rune {
```

File: src/subcommand/wallet/resume.rs [hunk 6]
```diff
@@ -18,7 +20,26 @@ impl Resume {
...
+
+        ensure!(
+          pending_etching.is_some(),
+          "rune `{spaced_rune}` does not correspond to any pending etching."
+        );
+
+        pending_etchings.push(pending_etching.unwrap());
+      } else {
+        pending_etchings.extend(wallet.pending_etchings()?);
+      }
```
Comment (raphjaph) [inline src/subcommand/wallet/resume.rs:40]:
```suggestion
        vec![pending_etching.unwrap()]
      } else {
        wallet.pending_etchings()?
      };
```

File: src/subcommand/wallet/resume.rs [hunk 7]
```diff
@@ -8,6 +8,8 @@ pub struct ResumeOutput {
 pub(crate) struct Resume {
   #[arg(long, help = "Don't broadcast transactions.")]
   pub(crate) dry_run: bool,
+  #[arg(long, value_name = "PENDING•RUNE", help = "Rune to resume (spaced).")]
```
Comment (raphjaph) [inline src/subcommand/wallet/resume.rs:11]:
```suggestion
  #[arg(long, help = "Rune to resume.")]
```

**Review by raphjaph — CHANGES_REQUESTED** [review]

**Review by raphjaph — CHANGES_REQUESTED** [review]
Almost there! Still needs an integration test. Have a look at `tests/wallet/resume.rs`. For reasons this can only be tested on Mac and Linux. What system are you on?

**Comment by ldiego08** [comment]
> Almost there! Still needs an integration test. Have a look at `tests/wallet/resume.rs`. For reasons this can only be tested on Mac and Linux. What system are you on?

@raphjaph took me a while to get around to adding these tests, but here they are! Hope it's looking good. I'm on a Mac, btw.

**Review by raphjaph — APPROVED** [review]
Great, LGTM!

**Pull request merged**